app/views/users/index.html.erb

```diff
@@ -1 +1,17 @@
-<h1>Here is a list of all users</h1>
\ No newline at end of file
+<!DOCTYPE html>
+<html>
+  <head></head>
+  <body>
+    <div class='container'>
+      <% @users.each do |user| %>
+        <div class='card'>
+          <img class='user_pic' src= <%= user.photo %> >
+          <a href="/users/<%= user.id %>" class='left_card'>
+            <h2 class='username'><%= user.name %></h2>
+            <p class='count'>Number of posts: <%= user.posts_counter >= 0 ? user.posts_counter : '0' %></p><br>
+          </a>
+        </div>
+      <% end %>
+    </div>
+  </body>
```

Comment:
- [OPTIONAL] Great job so far 👏🏻  However You may use Partial templates eg:  `app/views/users/_user.html.erb`, as a better practice instead of using `@objects.each`. Partial templates (partials) are a way of breaking the rendering process into more manageable chunks. Partials allow you to extract pieces of code from your templates to separate files and also reuse them throughout your templates. For more documentation visit this [link](https://riptutorial.com/ruby-on-rails/example/2875/partials)



```ruby
    # app/views/users/_user.html.erb

     #   HTML Code Here displaying
     #   user.photo, user.name and user.posts_counter

    # app/views/users/index.html.erb

    <%= render @users %>
```

Reply:
Oka thanks, a good suggestion I will make changes in the weak end for optional changes.